sdk/python/kfp/components/_components.py

```diff
@@ -216,16 +225,14 @@ def _create_task_spec_from_component_and_arguments(
             serialized_argument_value = serialize_value(argument_value, input_type)
             task_arguments[input_name] = serialized_argument_value
 
-    task = TaskSpec(
-        component_ref=component_ref,
-        arguments=task_arguments,
-    )
+    task = copy.copy(task_spec)
```

Comment:
IIUC this (and further modification to `task`) is almost an in-place modification to `task_spec`. I am wondering if it is possible to directly assign `task_spec.arguments = task_arguments` and work with `task_spec` directly. Did I miss anything here?

Reply:
The `task_spec` is given to this function from the outside. It might be a good idea not to make modifications to it so that there are no unexpected side effects of calling functions. This is why I usually make copies of passed in objects if I want to modify them.

Reply:
Then do we actually want `copy.deepcopy()`, which sounds safer to me?

Reply:
Well, we only make changes to the task_spec object itself, not any of its attributes, so there should not be a difference.